File: Matchismo 2/Matchismo/ViewController.m
//
//  ViewController.m
//  Matchismo
//
//  Created by Sanjay Bhatia on 10/29/14.
//  Copyright (c) 2014 Sanjay Bhatia. All rights reserved.
//

#import "ViewController.h"
#import "PlayingCardDeck.h"
#import "PlayingCard.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *flipsLabel;
@property (nonatomic) int flipCount;

@property (strong, nonatomic) Deck *myDeck;

@end

@implementation ViewController

- (Deck *) myDeck {
    if (! _myDeck) {
        _myDeck = [[PlayingCardDeck alloc] init];
    }
    return _myDeck;
}


- (IBAction)touchCardButton:(UIButton *)sender {

    if ([sender.currentTitle length]) {
        [sender setBackgroundImage:[UIImage imageNamed:@"cardback"]
                          forState:UIControlStateNormal];
        [sender setTitle:@"" forState:UIControlStateNormal];
        self.flipCount++;
    }
    else {
        Card *myCard = [self.myDeck drawRandomCard];
        
        if (myCard) {
            [sender setBackgroundImage:[UIImage imageNamed:@"cardfront"]
                              forState:UIControlStateNormal];
            [sender setTitle:myCard.contents forState:UIControlStateNormal];
            self.flipCount++;
        }
    }    
}


- (void) setFlipCount:(int)flipCount {
    _flipCount = flipCount;
    self.flipsLabel.text = [NSString stringWithFormat:@"Flips: %d", self.flipCount];
    NSLog(@"flipCount changed to %d", self.flipCount);
}



- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Calculator/Calculator/ViewController.swift
//
//  ViewController.swift
//  Calculator
//
//  Created by Sanjay Bhatia on 1/25/15.
//  Copyright (c) 2015 Sanjay Bhatia. All rights reserved.
//

import UIKit

class ViewController: UIViewController
{
    
    @IBOutlet weak var display: UILabel!
    var userIsInTheMiddleOfTypingANumber = false
    
    
    @IBAction func appendDigit(sender: UIButton) {
        
        let digit = sender.currentTitle!
        //println("Digit = \(digit)")
        
        if (userIsInTheMiddleOfTypingANumber) {
        display.text = display.text!  + digit
        }
        else {
            userIsInTheMiddleOfTypingANumber = true
            display.text = digit;
        }
        
    }

    @IBAction func enter() {
        userIsInTheMiddleOfTypingANumber = false
    
    }
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

